src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlAttributeCollection.cs

```diff
@@ -32,15 +34,15 @@ internal XmlAttributeCollection(XmlNode parent) : base(parent)
 
         // Gets the attribute with the specified name.
         [System.Runtime.CompilerServices.IndexerName("ItemOf")]
-        public XmlAttribute this[string name]
+        public XmlAttribute? this[string name]
         {
             get
             {
                 int hash = XmlName.GetHashCode(name);
 
                 for (int i = 0; i < nodes.Count; i++)
                 {
-                    XmlAttribute node = (XmlAttribute)nodes[i];
+                    XmlAttribute node = (XmlAttribute)nodes[i]!;
```

Comment:
I made it nullable because XmlNamedNodeMap.SmallXmlNodeList has a special case for adding null and then internally adds null itself to underlying array when that happens which made me assume it accepts null and can return null. Looking at the usages though I don't see anything expecting to return null though so I'll add Debug.Fail for that special case and see if it ever gets triggered (at least we won't break any Release code if someone already does it in an existing app)